### PR TITLE
docs: update RadioCardGroup to v2 docs

### PR DESCRIPTION
**Pull request by scurker**

### CODE REVIEW

File: docs/pages/components/RadioCardGroup.mdx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+## Props
+
+<ComponentProps
+  className={true}
+  props={[
+    {
+      name: 'radios',
+      type: 'Array',
+      required: true,
+      description: 'Array of objects containing: label, cardImg, cardIcon (see IconTypes), value (optional), and any supported HTMLInputElement properties'
```
Comment (dequejosie) [inline docs/pages/components/RadioCardGroup.mdx:90]: nit: period.

File: docs/pages/components/RadioCardGroup.mdx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      description: 'A label for the radio card group is required. This means you must provide either an aria-label or aria-labelledby prop.',
+    },
+    {
+      name: 'aria-labelledby',
+      type: 'string',
+      description: 'A label for the radio card group is required. This means you must provide either an aria-label or aria-labelledby prop.',
+    },
+    {
+      name: 'name',
+      type: 'string'
```
Comment (dequejosie) [inline docs/pages/components/RadioCardGroup.mdx:104]: what is this `name` here?

**Comment by dequejosie** [comment]
Not so much about the docs, the dark theme of disabled RadioCardGroup looks weird to me. It does not have any visual indication that the card is disabled:
Screen Shot 2023-08-22 at 4 50 29 PM

And it looks like this during hover:
Screen Shot 2023-08-22 at 4 50 37 PM

**Comment by scurker** [comment]
> Not so much about the docs, the dark theme of disabled RadioCardGroup looks weird to me. It does not have explicit visual indication that the card is disabled:

Correct. This is a documentation ticket so it doesn't implement any fixes. There is a ticket to address some of the a11y issues with RadioCardGroup (https://github.com/dequelabs/cauldron/issues/777) but it's low priority since this component is currently only used in one spot that doesn't have themes.

File: docs/pages/components/RadioCardGroup.mdx (hunk 2 above)
Comment (scurker) [reply docs/pages/components/RadioCardGroup.mdx]: I was copying over the existing documentation and it wasn't there, my bad.

**Review by thuey — APPROVED** [review]

**Pull request merged**